src/lib.rs

```diff
@@ -32,10 +80,16 @@ pub fn generate_hashes_for_string(
         as_u32_be(&sha256_hash.result().into())
     };
 
-    make_tri_grams(s)
+    let result: HashSet<_> = make_tri_grams(s)
         .iter()
         .map(|tri_gram| short_hash(tri_gram.as_bytes()))
-        .collect()
+        .take(MAX_HASHES_SIZE + 1) //Take 1 more than the max so we can detect it. Since this is lazy it could save us work.
+        .collect();
+    if result.len() > MAX_HASHES_SIZE {
```

Comment:
I think the length limit should be on the number of chars in the input string - that automatically gives us a limit on the number of trigrams as well, and it is easier to explain to people (and to check in a UI).

Reply:
Wouldn't that be difficult with spaces in the input? 
"1234" is 4 characters and 2 trigrams.
"123 456" is 7 characters and also 2 trigrams.
Someone with many names could have a long input string even though it results in few trigrams.

Reply:
The maximum number of trigrams a string of n chars can produce is n-2, but it could be shorter. So we automatically get a limit on the number of tokens based on the input. We don't care if it is shorter - we are going to add some padding anyway.

Reply:
I discussed this with Bob and I moved to checking the string length. It makes the error conditions easier to understand for the layperson.